### ISSUE TITLE
Additional Lances: Spawn lances with battle damage

### ISSUE THREAD

**Issue by NickAragua** [CWolfs/MissionControl#76]
The base game spawns lances that are at 25, 50 or 75% armor, but the lances spawned by Mission Control are always at full health. Combined with the treatment of the spawned lances as primary objectives in battle missions, it makes those missions basically impossible. 

So, is it possible to emulate the base game's mechanism of deploying enemy forces with limited armor?

**Comment by CWolfs** [comment]
I actually had this feature in v0.1.0 but assumed players would prefer full health allies and enemies to fight against.

I'll add something like this back in. 

While you wait you're able to add this to your version yourself. If you add the appropriate spawn tags to the generic lance in the Mission Control folder then they will spawn with less armour. I'm away this weekend but I'll try to give you a full example on what to do soon.

**Comment by NickAragua** [comment]
If I only add the damaged battle lance to the General.json, do I also need to add it to each of the "difficultyx.json" files? i.e. which one takes precedence?

**Comment by CWolfs** [comment]
If MCs setting.json has SkullValueMatters as true (true by default) then when you go into a contract the skull value will determine which Difficulty file to use. If it can't find one (e.g a player or a mod maker decides to delete one for a reason) then it falls back to the general.json.

Skull value goes from 1/2 skull (difficulty 1) up to 5 skulls (difficulty 10).

With the v0.2.0 prerelease alpha all the difficulty profiles are the same as I didn't get time to configure them before BT v1.3 was released.

**Comment by NickAragua** [comment]
Gotcha. I toned it down a little for the time being (to 1 max extra enemy lance and .2 chance of allied lance) as having two lances spawn on top of my guys in addition to whatever the mission throws in was a bit much. Unfortunately, I think there's a "soft" but fairly punishing limit for how much extra stuff should be spawned on a mission, given that the maps are not very big in general.

I'm probably going to take some time to fool around with configuring lance settings for different battle types, since that appears to be an option. Do we have access to the specific mission name also, or just the type of battle?

**Comment by CWolfs** [comment]
Glad you found out how to get it closer to what you'd like. Sadly, I agree. The small maps do limit the amount of lances that can spawn purely from a gameplay perspective. Future versions of Mission Control might be able to increase the size of the map but there's a hard limit to how far I can go due to how the game handles the maps. Basically a 2k x 2k play area.

Have fun playing with options. The main reason I made things so configurable is because I knew different people would want different things from Mission Control - so I gave them a way to do it themselves.

I've intentionally avoided control of Mission Control down to the specific mission at the moment as it makes things more future proof to handle it at a contract type level. I've had some requests to be able to manipulate specific missions though so I'm thinking how best to implement this in a future proof manner.
